src/main/java/uk/org/tombolo/DataExportRunner.java

```diff
@@ -1,113 +1,121 @@
-        if (!report.isSuccess()) {
-            DataExportRecipeValidator.display(report);
-            System.exit(1);
+    /**
+     * Performs correlation analysis.
+     *
+     * Calculates the correlation between all input fields and output the Pearson correlation coefficient,
+     * the pValue and the standard error to a JSON file.
+     *
+     * Currently we only support the use of GeoJson as the intermediate data export format.
+     * We chose this since it is a cleaner implementation than our CSV exporter
+     * and the additional benefit is to be able to visualise the intermediate data in QGIS.
+     *
+     * @param dataExportRecipe Data export recipe
+     * @param output File containing the exported fields to use in the correlation analysis. We get this file by running
+     *               the recipe.
+     * @param correlation Output file for the correlation results
+     * @throws Exception
+     */
+    private void runCorrelation(DataExportRecipe dataExportRecipe, String output, String correlation) throws Exception {
+        List<FieldRecipe> fields = dataExportRecipe.getDataset().getFields();
+
+        // Read in data file
+        RealMatrix matrix;
+        if (dataExportRecipe.getExporter().equals(GeoJsonExporter.class.getCanonicalName())){
+            matrix = CorrelationAnalysisEngine.readGeoJsonDataExport(output, fields);
+        }else if(dataExportRecipe.getExporter().equals(CSVExporter.class.getCanonicalName())){
+            matrix = CorrelationAnalysisEngine.readCSVDataExport(output, fields);
+        }else {
+            throw new Error("Unknown exporter class for intermediate data.");
         }
-    }
 
-    private static void validateArguments(String[] args) {
-        if (args.length != 5){
-            log.error("Use: {} {} {} {} {}",
-                    DataExportRunner.class.getCanonicalName(),
-                    "dataExportSpecFile",
-                    "outputFile",
-                    "clearDatabaseCache",
-                    "forceImports (className:datasourceId,...)",
-                    "dataExportSpec"
-            );
-            System.exit(1);
-        }
+        // Calculate and output correlations
+        CorrelationAnalysisEngine.calculateAndOutputCorrelations(matrix, fields, correlation);
     }
 
-    private static Writer getOutputWriter(String path) {
-        try {
-            return new FileWriter(path);
-        } catch (IOException e) {
-            log.error("Error initialising output writer: {}", path);
+    @Override
+    protected void validateArguments(String[] args) {
+        if (args.length != 6) {
+            log.info("Use: {} {} {} {} {}",
+                    "s",
+                    "recipe",
+                    "output",
+                    "correlation",
+                    "force (className:datasourceId,...)",
+                    "clear"
+            );
             System.exit(1);
```

Comment:
We could actually completely abstract it, by putting two more arguments ```int lengthRequired``` and ```String errorMsg```. Just a suggestion if you would like

Reply:
Actually we could even remove this, as it's never hit. In gradle we make sure to have 6 arguments. If less than 6, because the mandatory ones are missing, gradle will fail before reaching it.